### PR TITLE
Adds ManagedIndexService/PolicyService to public services, updates ba…

### PR DESCRIPTION
**Pull request by dbbaughe**
…ckend services and types

*Issue #, if available:*

*Description of changes:*
Adds ManagedIndexService/PolicyService to public services, updates ba…

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: public/services/IndexService.ts [hunk 1]
```diff
@@ -17,6 +17,7 @@ import { IHttpResponse, IHttpService } from "angular";
 import { INDEX } from "../../server/utils/constants";
 import { AddPolicyResponse, GetIndicesResponse, SearchResponse } from "../../server/models/interfaces";
 import { ServerResponse } from "../../server/models/types";
+import { NODE_API } from "../../utils/constants";
```
Comment (seraphjiang) [inline public/services/IndexService.ts]: nit: prefer constants under its own folder/module

File: public/services/IndexService.ts [hunk 2]
```diff
@@ -26,25 +27,25 @@ export default class IndexService {
   }
 
   getIndices = async (queryParamsString: string): Promise<ServerResponse<GetIndicesResponse>> => {
-    const response: IHttpResponse<ServerResponse<GetIndicesResponse>> = await this.httpClient.get(
-      `../api/ism/_indices?${queryParamsString}`
-    );
+    const url = `..${NODE_API._INDICES}?${queryParamsString}`;
+    const response = (await this.httpClient.get(url)) as IHttpResponse<ServerResponse<GetIndicesResponse>>;
```
Comment (seraphjiang) [inline public/services/IndexService.ts]: nit: no need to create local variable for response unless we will use it, e.g. logging
we could return {}.data directly.

Comment (dbbaughe) [reply public/services/IndexService.ts]: I generally agree, I just find this hard to read:
```
return ((await this.httpClient.get(url)) as IHttpResponse<ServerResponse<GetIndicesResponse>>).data;
```

File: public/services/IndexService.ts (hunk 1 above)
Comment (dbbaughe) [reply public/services/IndexService.ts]: We can discuss outside this PR to see what we will adopt across all plugins and take action after, don't want to spam this PR with moving constants to its own module if we do do that.

File: public/services/ManagedIndexService.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    return response.data;
+  };
+
+  getManagedIndices = async (queryParamsString: string): Promise<ServerResponse<GetManagedIndicesResponse>> => {
+    const response = (await this.httpClient.get(`..${NODE_API.MANAGED_INDICES}?${queryParamsString}`)) as IHttpResponse<
+      ServerResponse<GetManagedIndicesResponse>
+    >;
+    return response.data;
+  };
+
```
Comment (seraphjiang) [inline public/services/ManagedIndexService.ts]: wondering if we could have private generic getmethod
getData<InputType, InputParamType>(params: string)

File: server/services/PolicyService.ts [hunk 4]
```diff
@@ -83,17 +83,17 @@ export default class PolicyService {
...
       const { callWithRequest } = await this.esDriver.getCluster(CLUSTER.ISM);
       const getResponse = await callWithRequest(req, "ism.getPolicy", params);
       const policy = _.get(getResponse, "policy", null);
-      const seqNo = _.get(getResponse, "_seq_no", null);
-      const primaryTerm = _.get(getResponse, "_primary_term", null);
+      const seqNo = _.get(getResponse, "_seq_no");
+      const primaryTerm = _.get(getResponse, "_primary_term");
       if (policy) {
-        return { ok: true, response: { id, seqNo, primaryTerm, policy } };
+        return { ok: true, response: { id, seqNo: seqNo as number, primaryTerm: primaryTerm as number, policy: policy as Policy } };
```
Comment (seraphjiang) [inline server/services/PolicyService.ts]: nit: maybe just +seqNo for convert to number
https://stackoverflow.com/questions/14667713/typescript-converting-a-string-to-a-number

**Comment by dbbaughe** [comment]
Will resolve nitpicks in a later PR, thanks!

**Pull request merged**